01-messaging.md

```diff
@@ -153,6 +156,73 @@ diagnosis, as it indicates that one peer has a bug.
 It may be wise not to distinguish errors in production settings, lest
 it leak information, thus the optional data field.
 
+## Control Messages
+
+### The `ping` message
+
+In order to allow for the existence of very long-lived TCP connections, at
+times it may be required that both ends keep alive the TCP connection at the
+application level.
+
+1. type: 18 (`ping`)
+2. data: 
+    * [2:num_pong_bytes]
+
+#### Requirements
+
+A node receiving a `ping` message MUST respond with a `pong` message that
+contains the specified `num_pong_bytes` number of bytes within it's data
```

Comment:
Would it be a good idea to also put padding data in the `ping` message for further obfuscation ? Otherwise eavesdroppers might guess a `ping` has been sent?